sys/fs/squashfs/squashfs_inode.c

```diff
@@ -226,6 +282,159 @@ sqsh_init_lreg_inode(struct sqsh_mount *ump, struct sqsh_inode *inode)
 	return (SQFS_OK);
 }
 
+sqsh_err
+sqsh_init_dir_inode(struct sqsh_mount *ump, struct sqsh_inode *inode)
+{
+	struct sqsh_dir_inode temp;
+	sqsh_err err;
+
+	err = sqsh_metadata_get(ump, &inode->next, &temp, sizeof(temp));
+	if (err != SQFS_OK)
+		return (err);
+	swapendian_dir_inode(&temp);
+
+	/* initialise inode dir fields */
+	inode->nlink					=	temp.nlink;
+	inode->xtra.dir.start_block 	=	temp.start_block;
+	inode->xtra.dir.offset			=	temp.offset;
+	inode->size						=	temp.file_size;
+	inode->xtra.dir.idx_count		=	0;
+	inode->xtra.dir.parent_inode	=	temp.parent_inode;
+
+	return (SQFS_OK);
+}
+
+sqsh_err
+sqsh_init_ldir_inode(struct sqsh_mount *ump, struct sqsh_inode *inode)
+{
+	struct sqsh_ldir_inode temp;
+	sqsh_err err;
+
+	err = sqsh_metadata_get(ump, &inode->next, &temp, sizeof(temp));
+	if (err != SQFS_OK)
+		return (err);
+	swapendian_ldir_inode(&temp);
+
+	/* initialise inode dir fields */
+	inode->nlink					=	temp.nlink;
+	inode->xtra.dir.start_block 	=	temp.start_block;
+	inode->xtra.dir.offset			=	temp.offset;
+	inode->size						=	temp.file_size;
+	inode->xtra.dir.idx_count		=	temp.i_count;
+	inode->xtra.dir.parent_inode	=	temp.parent_inode;
+	inode->xattr					=	temp.xattr;
+
+	return (SQFS_OK);
+}
+
+sqsh_err
+sqsh_init_symlink_inode(struct sqsh_mount *ump, struct sqsh_inode *inode)
+{
+	struct sqsh_symlink_inode temp;
+	sqsh_err err;
+
+	err = sqsh_metadata_get(ump, &inode->next, &temp, sizeof(temp));
+	if (err != SQFS_OK)
+		return (err);
+
+	/* initialise inode dir fields */
+	inode->nlink				=	le32toh(temp.nlink);
+	inode->size					=	le32toh(temp.symlink_size);
+
+	return (SQFS_OK);
+}
+
+sqsh_err
+sqsh_init_dev_inode(struct sqsh_mount *ump, struct sqsh_inode *inode)
+{
+	struct sqsh_dev_inode temp;
+	sqsh_err err;
+
+	err = sqsh_metadata_get(ump, &inode->next, &temp, sizeof(temp));
+	if (err != SQFS_OK)
+		return (err);
+
+	/* set size of inode data to 0 */
```

Comment:
Comments that describe simple assignments don't add to the clarity of the code and could be left out.  You might consider deleting the white space between the 3 assignments as well to make more of the function fit on a screen.

Reply:
Done.
I deleted other similar assignment comments as well.